ADR/0008-integration-service-api-contracts.md

```diff
@@ -0,0 +1,277 @@
+
+It is advisable to limit the number of structured fields as it impacts indexing. Fields that are common to multiple tasks should be preferred. Before a field specific to a task is added it is worth thinking whether it is really needed and whether the same information can be conveyed through a more generic field. Information that does not need to be “searchable” can be recorded in an unstructured way.
+
+### Results and Records
+
+Related to [PLNSRVCE-41](https://issues.redhat.com/browse/PLNSRVCE-41) investigations in regards to [Tekton results](https://github.com/tektoncd/results). The investigations are documented [here](https://docs.google.com/document/d/1WGWx6MllVs-BwRLd0PP5QTQHgGp0fW2dOYRKRAKWiwo/edit#heading=h.bpyxfhyud4dv).
+
+The output of each Tekton task will be provided in two forms:
+- As a minimized [Tekton result](https://tekton.dev/docs/pipelines/tasks/#emitting-results) in JSON format listing all test failures
+- The name of the result will be **HACBS_TEST_OUTPUT**
+- As a Test output JSON file with full test information that’s saved in the Tekton Pipeline Workspace
+  - The name of the file will be in snake case and will be in the form of **test_name_output.json**
+
+##### Tekton Results
+
+Each test will have a standardized short name in snake case, e.g. `release_label_required`, `architecture_label_deprecated` etc.
+
+The Test output of the Tekton result will be a JSON object that includes the context about the test along with the list of check names for all failed checks
+
+The output will provide the following information about the overall test result:
+- **result** - The outcome of the testing task, can be `SUCCESS`, `FAILURE`, `WARNING`, `SKIPPED` or `ERROR`
+- **namespace** - The rego namespace of the test policy
+- **timestamp** - An UNIX epoch timestamp of the test completion time
+- **successes** - The number of successful checks in the form of an integer
+- **note** - A short note providing additional information about the test
+- **failures** - The number of failed checks in the form of an integer
+- **warnings** - The number of warning checks in the form of an integer
+
+Example contents of the test result output file (**HACBS_TEST_OUTPUT**) for a failed run:
+```
+{ 
+    “result”: “FAILURE”,
+    “namespace”: “image_labels”,
+    “timestamp”: “1649148140”,
+    “successes”: 12, 
+    “note”: “”,
+    “failures”: 2
+}
+```
+Example for a successful run:
+```
+{
+    "result": "SUCCESS",
+    "timestamp":"1649843611",
+    "namespace":"required_checks",
+    "successes":16,
+    “note”: “”,
+    "failures": 0
+}
+```
+Example for a skipped run:
+```
+{
+    "result":"SKIPPED",
+    “note”: “<skipped message>”,
+    "timestamp":"1649842004"
+}
+```
+Example for a run with an error:
+```
+{
+    "result":"ERROR",
+    "timestamp":"1649842004"
+}
+```
+
+###### Proposed Adapted Tekton Results for Xunit Tests Results
```

Comment:
Xunit test result has a well-knowen format that is understood by a lot of tools. I don't think we should transform it into our own format. We can just forward it as is.

Reply:
Yeah, the idea wasn't to completely convert it and display the whole junit/xunit results output. I agree it is a well understood format. So full details it can be viewed elsewhere. The idea was to provide a summarized of the results similar to what we did with the Sanity/SAST tasks for specific tekton task that was go to execute the test framework. Figured I would provide some context since I originally added that into the contract doc :)

Reply:
Thanks @Dannyb48, I feel it is a summary as well so we can get the points quickly.

Reply:
Yeah - I think this is the crux of the ADR. We have to get results out of a task by way of HACBS_TEST_OUTPUT because of the size limitation on task results. Normal results (like full xunit results) just can't be exposed.